### ISSUE TITLE
Dockerfile will not build images

### ISSUE THREAD

**Issue by qfettes** [radeklat/todoist-habitica-sync#3]
The current image on dockerhub is for x86 architectures, and will not work on arm based architectures (a raspberry pi is a great device to run your tool continuously). Using the dockerfile to build a new image did not work due to a bug with the alpine3.13 base image. There is a quick, 2-line fix for the Dockerfile which will allow it to build correctly. If you'd like, you can make the change such that the first lines of the dockerfile look like this:

```
FROM python:3.9-alpine3.12

RUN python3 -m pip install --upgrade pip
```

This fix utilizes a different alpine linux image, and adds a command to upgrade pip in the Dockerfile. This allows a new image to be built on arm based architectures. Or, you could give me permission to branch and make a pull request.

**Comment by radeklat** [comment]
Hey @qfettes. I added a support for building multiple platforms into the CD pipeline. Unfortunately it fails to build for `linux/arm/v7`. Your suggestion didn't fix the error I'm seeing. I'll try to investigate how to solve it. But if you want to give it a try as well, the build command to reproduce it from the repository root is

```bash
docker buildx build --platform linux/amd64,linux/arm/v7 .
```

At the moment it raises:

```
------                                                                                                                
 > [linux/arm/v7 2/8] RUN apk update   && apk add --no-cache bash openssl-dev gcc libffi-dev cargo   && rm -rf /var/lib/apt/lists/*:                                                                                                        
#7 0.196 fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/main/armv7/APKINDEX.tar.gz                                 
#7 0.937 fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/community/armv7/APKINDEX.tar.gz                            
#7 1.570 v3.13.5-218-g69d8716d3d [https://dl-cdn.alpinelinux.org/alpine/v3.13/main]                                   
#7 1.570 v3.13.5-216-gca846067ce [https://dl-cdn.alpinelinux.org/alpine/v3.13/community]
#7 1.570 OK: 13500 distinct packages available
#7 1.657 fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/main/armv7/APKINDEX.tar.gz
#7 2.267 fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/community/armv7/APKINDEX.tar.gz
#7 2.762 (1/29) Upgrading musl (1.2.2-r0 -> 1.2.2-r1)
#7 2.836 (2/29) Installing bash (5.1.0-r0)
#7 2.950 Executing bash-5.1.0-r0.post-install
#7 2.953 ERROR: bash-5.1.0-r0.post-install: script exited with error 1
#7 2.953 (3/29) Installing rust-stdlib (1.47.0-r2)
#7 7.193 (4/29) Installing libstdc++ (10.2.1_pre1-r3)
#7 7.322 (5/29) Installing binutils (2.35.2-r1)
#7 7.598 (6/29) Installing libgomp (10.2.1_pre1-r3)
#7 7.642 (7/29) Installing libatomic (10.2.1_pre1-r3)
#7 7.669 (8/29) Installing libgphobos (10.2.1_pre1-r3)
#7 7.886 (9/29) Installing gmp (6.2.1-r0)
#7 7.946 (10/29) Installing isl22 (0.22-r0)
#7 8.049 (11/29) Installing mpfr4 (4.1.0-r0)
#7 8.108 (12/29) Installing mpc1 (1.2.0-r0)
#7 8.144 (13/29) Installing gcc (10.2.1_pre1-r3)
#7 11.56 (14/29) Installing musl-dev (1.2.2-r1)
#7 11.83 (15/29) Installing libxml2 (2.9.10-r7)
#7 11.90 (16/29) Installing llvm10-libs (10.0.1-r1)
#7 14.45 (17/29) Installing rust (1.47.0-r2)
#7 18.64 (18/29) Installing brotli-libs (1.0.9-r3)
#7 18.70 (19/29) Installing nghttp2-libs (1.42.0-r1)
#7 18.73 (20/29) Installing libcurl (7.77.0-r1)
#7 18.81 (21/29) Installing http-parser (2.9.4-r0)
#7 18.84 (22/29) Installing pcre (8.44-r0)
#7 18.91 (23/29) Installing libssh2 (1.9.0-r1)
#7 18.97 (24/29) Installing libgit2 (1.1.0-r1)
#7 19.09 (25/29) Installing cargo (1.47.0-r2)
#7 19.60 (26/29) Installing linux-headers (5.7.8-r0)
#7 19.82 (27/29) Installing pkgconf (1.7.3-r0)
#7 19.86 (28/29) Installing libffi-dev (3.3-r2)
#7 19.89 (29/29) Installing openssl-dev (1.1.1k-r0)
#7 19.99 Executing busybox-1.32.1-r5.trigger
#7 19.99 ERROR: busybox-1.32.1-r5.trigger: script exited with error 1
#7 20.00 1 error; 300 MiB in 64 packages
------
Dockerfile:6
--------------------
   5 |     
   6 | >>> RUN apk update \
   7 | >>>   && apk add --no-cache bash openssl-dev gcc libffi-dev cargo \
   8 | >>>   && rm -rf /var/lib/apt/lists/*
   9 |     
--------------------
error: failed to solve: rpc error: code = Unknown desc = executor failed running [/dev/.buildkit_qemu_emulator /bin/sh -c apk update   && apk add --no-cache bash openssl-dev gcc libffi-dev cargo   && rm -rf /var/lib/apt/lists/*]: exit code: 1
```

So far, I managed to narrow the error down to be coming from `apk add --no-cache bash` (or I presume anything else).

If you want to dig in and try to find a fix it would be much appreciated.

**Comment by qfettes** [comment]
I took a quick look today and couldn't find a solution. The error seems to affect most of the packages (only apk add --no-cache openssl-dev works) in the apk add statement, and also occurs with every version of alpine linux I tried. It also only affects the arm build.

I'll take a second look when I have time this week. I do see a few similar issues open around github: https://github.com/docker/buildx/issues/508
